### PR TITLE
Add a new lambda functions called EBS janitor

### PR DESCRIPTION
**Pull request by spirosoik**
#### Summary
The EBS janitor is responsible to delete volumes which are not in use and they have been
created very long time ago (specifically default configuration is 90 days). If a volume
is available and created more than 90days (~3 months) we delete the volume for cleaning up
and keep the cost down.

#### Ticket Link
Ticket: https://mattermost.atlassian.net/browse/MM-37523

### CODE REVIEW

**Review by stafot — APPROVED** [review]
LGTM.

**Comment by stafot** [comment]
Needs some extra logic on CI. ref. https://github.com/mattermost/mattermost-cloud-monitoring/pull/255/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47 cc @spirosoik

**Review by stafot — APPROVED** [review]

File: lambda-functions/ebs-janitor/event_handler.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			continue
+		}
+		h.logger.WithFields(fields).Info("volume to be deleted")
+		ctx, cancel = context.WithTimeout(context.Background(), awsTimeout)
+		defer cancel()
+		if h.dryRun {
+			continue
+		}
+		if err := h.awsResourcer.DeleteVolume(ctx, v.VolumeId); err != nil {
+			h.logger.WithFields(fields).Info("failed to delete volume")
```
Comment (stylianosrigas) [inline lambda-functions/ebs-janitor/event_handler.go:63]: It should be .Error right?

File: lambda-functions/ebs-janitor/main.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	})
+	if err != nil {
+		log.WithError(err).Error("failed initiate an AWS session")
+		return
+	}
+	// setup the handler
+	awsResourcer := NewClient(sess)
+	handler := NewEventHandler(cfg.ExpirationDays, awsResourcer, cfg.Debug, logger)
+	if cfg.Debug {
+		handler.Handle(context.Background(), events.CloudWatchEvent{}) //nolint
```
Comment (stylianosrigas) [inline lambda-functions/ebs-janitor/main.go:50]: Don't we need to get the error here?

**Review by stylianosrigas — COMMENTED** [review]
Just some minor comments.

File: lambda-functions/ebs-janitor/main.go (hunk 2 above)
Comment (spirosoik) [reply lambda-functions/ebs-janitor/main.go:50]: we can but we don't care about the error as will crash here is only for debugging locally actually.

File: lambda-functions/ebs-janitor/event_handler.go (hunk 1 above)
Comment (spirosoik) [reply lambda-functions/ebs-janitor/event_handler.go:63]: right, let me fix this

**Review by stylianosrigas — APPROVED** [review]

**Pull request merged**